pkg/vsphere/disk/disk_manager.go

```diff
@@ -46,17 +46,28 @@ type Manager struct {
 	// reference to the vm this is running on.
 	vm *vm.VirtualMachine
 
+	// VirtualDiskManager that is used to create vmdks directly on datastore
+	// from https://pubs.vmware.com/vsphere-65/index.jsp?topic=%2Fcom.vmware.vspsdk.apiref.doc%2Fvim.VirtualDiskManager.html
+	// Most VirtualDiskManager APIs will be DEPRECATED as of vSphere 6.5. Please use VStorageObjectManager APIs to manage Virtual disks.
+	vdm *object.VirtualDiskManager
+
 	// The controller on this vm.
 	controller *types.ParaVirtualSCSIController
 
 	// The PCI + SCSI device /dev node string format the disks can be attached with
 	byPathFormat string
 
-	reconfig sync.Mutex
+	// serialize reconfigure operations
+	mu sync.Mutex
+
+	// map of URIs to VirtualDisk structs so that we can return the same instance to the caller, required for ref counting
+	Disks map[uint64]*VirtualDisk
```

Comment:
How are we populating this with existing disks - walk stores on start or lazily?